antarest/study/storage/rawstudy/raw_study_service.py

```diff
@@ -402,11 +402,38 @@ def unarchive(self, study: RawStudy) -> None:
         Raises:
             BadArchiveContent: If the archive is corrupted or in an unknown format.
         """
-        with open(self.get_archive_path(study), mode="rb") as fh:
+        with open(self.find_archive_path(study), mode="rb") as fh:
             self.import_study(study, fh)
 
-    def get_archive_path(self, study: RawStudy) -> Path:
-        return Path(self.config.storage.archive_dir / f"{study.id}.zip")
+    def find_archive_path(self, study: RawStudy) -> Path:
+        """
+        Fetch for archive path of a study if it exists else raise an incorrectly archived study.
+
+        Args:
+            study: The study to get the archive path for.
+
+        Returns:
+            The full path of the archive file (zip or 7z).
+        """
+        archive_dir: Path = self.config.storage.archive_dir
+        for suffix in [".zip", ".7z"]:
+            path = archive_dir.joinpath(f"{study.id}{suffix}")
+            if path.is_file():
+                return path
+        raise FileNotFoundError(f"Study {study.id} archiving process is corrupted (no archive file found).")
+
+    def infer_archive_path(self, study: RawStudy) -> Path:
```

Comment:
should be private. Also "infer" sounds like we try to guess something complex, why not just "get" ?

Reply:
As I Explained to @MartinBelthle, the name of this function was proposed by Laurent.
It was `create_archive_path`, then Martin came with another name to finally stick to this.

However, I am going to figure out a way to just not use it.